### PR TITLE
Remove session tracking in proxy mode

### PR DESCRIPTION
**Pull request by hennevogel**
This was an attempt to measure logins with proxy authentication based
on User.session. But User.session is a thread variable that is reset
after each request. So it's basically impossible to check based on this.
The user migh just have authenticated or we might just be on a new
thread ¯\_(ツ)_/¯

It also doesn't make much sense to measure :unauthenticated and
:disabled without measuring the rest of the case.

So let's stop measuring this at all until we find a better way.

For non-proxy auth this is still measured in Webui::SessionController.

### CODE REVIEW

File: src/api/app/controllers/webui/webui_controller.rb [hunk 1]
```diff
@@ -197,12 +195,6 @@ def check_user
     User.session = User.find_by_login(session[:login]) if session[:login]
 
     User.session ||= User.possibly_nobody
-
-    if !User.session
```
Comment (krauselukas) [inline src/api/app/controllers/webui/webui_controller.rb:201]: @hennevogel why did you remove this `if/else` block? doesn't seem to be related to the proxy mode

Comment (hennevogel) [reply src/api/app/controllers/webui/webui_controller.rb:201]: You are right, this code would also run without proxy mode. But it still would generate "wrong" data.

Once `User.session` is false we run an unauthenticated request, but that isn't data we want to track. And `previous_user` being different from `User.session` does *not* mean a log in happened. 

Let me split those to another commit...

Comment (hennevogel) [reply src/api/app/controllers/webui/webui_controller.rb:201]: Did not find a way to split this more intelligent so I made a new commit message. Please check again.

Comment (krauselukas) [reply src/api/app/controllers/webui/webui_controller.rb:201]: okay understand :+1:
